src/spatialdata/models/models.py

```diff
@@ -642,24 +643,77 @@ class TableModel:
     REGION_KEY_KEY = "region_key"
     INSTANCE_KEY = "instance_key"
 
-    def validate(
-        self,
-        data: AnnData,
-    ) -> AnnData:
+    def _validate_set_instance_key(self, data: AnnData, instance_key: str | None = None) -> None:
         """
-        Validate the data.
+        Validate the instance_key in table.uns or set a new instance_key as the instance_key column.
+
+        If no instance_key is provided as argument, the presence of instance_key in table.uns is checked and validated.
+        If instance_key is provided, presence in table.obs will be validated and if present it will be set as the new
+        instance_key in table.uns.
 
         Parameters
         ----------
         data
-            The data to validate.
+            The AnnData table.
+
+        instance_key
+            The instance_key to be validated and set in table.uns.
+
+        Raises
+        ------
+        ValueError
+            If no instance_key is provided as argument and no instance_key is found in the `uns` attribute of table.
+        ValueError
+            If no instance_key is provided and the instance_key in table.uns does not match any column in table.obs.
+        ValueError
+            If provided instance_key is not present as table.obs column.
```

Comment:
I think there is a missing case: both instance_key is provided and instance_key is already present in table.uns. I think we should raise an exception to the user.

Reply:
What would be the reason for raising the exception? The validation is a private one and is only used when changing the annotation target. So the intent is in general to overwrite if a value for instance_key is given.

Reply:
Ok makes sense!